### PR TITLE
Masthead dropdown links for subdomains

### PR DESCRIPTION
**Pull request by neoformit**
Currently, for galaxy servers using subdomains (e.g. `rna.usegalaxy.eu`) there is no way for users to easily switch between sites, or even see what sites are available. This PR adds an optional & configurable masthead/navbar dropdown for Galaxy subdomains/community domains site links. This masthead item is only displayed when a `subdomains_conf.xml` config file is present. Domain links matching the current `location.origin` are hidden. 

e.g.
image

## How to test the changes?
(Select all options that apply)
- [x] This is a refactoring of components with existing test coverage.

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/config/sample/subdomains_conf.yml.sample [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Each subdomain entry will be rendered as a link in the navbar
+# Ensure that you add an entry for your base site
+
+subdomains:
+  - title: Base
+    url: https://mygalaxy.org
```
Comment (martenson) [inline lib/galaxy/config/sample/subdomains_conf.yml.sample]: This is an existing and parked domain. 

```suggestion
    url: https://mygalaxy.example.com/
```

File: lib/galaxy/config/sample/subdomains_conf.yml.sample [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  - title: Genomics
+    url: https://genome.mygalaxy.org
```
Comment (martenson) [inline lib/galaxy/config/sample/subdomains_conf.yml.sample]:
```suggestion
    url: https://genome.example.com
```

**Review by martenson — COMMENTED** [review]
I am not sure "Domain" menu entry is going to communicate to the user what is happening here well enough. It is opt-in though...

**Comment by neoformit** [comment]
> I am not sure "Domain" menu entry is going to communicate to the user what is happening here well enough. It is opt-in though...

@martenson Good point, will review with our team. Maybe "Sites" is better. I suppose this feature could also be generalised as "external sites/links" and used to create links to any external site - would just make "Domain" a configurable in `subdomain_conf.yml` and change the naming like `subdomain` -> `external_sites`.

**Comment by neoformit** [comment]
@bgruening has suggested this can probably be done with a webhook plugin. Closing for now...

**Pull request closed without merging**